tests/sentry/api/endpoints/test_release_file_details.py

```diff
@@ -44,6 +44,48 @@ def test_simple(self):
         assert response.status_code == 200, response.content
         assert response.data['id'] == six.text_type(releasefile.id)
 
+    def test_file_download(self):
+        self.login_as(user=self.user)
+
+        project = self.create_project(name='foo')
+
+        release = Release.objects.create(
+            project=project,
+            organization_id=project.organization_id,
+            version='1',
+        )
+        release.add_project(project)
+
+        from io import BytesIO
+        f = File.objects.create(
+            name='application.js',
+            type='release.file',
+        )
+        f.putfile(BytesIO('File contents here'))
+
+        releasefile = ReleaseFile.objects.create(
+            organization_id=project.organization_id,
+            project=project,
+            release=release,
+            file=f,
+            name='http://example.com/application.js'
+        )
+
+        url = reverse('sentry-api-0-release-file-details', kwargs={
+            'organization_slug': project.organization.slug,
+            'project_slug': project.slug,
+            'version': release.version,
+            'file_id': releasefile.id,
+        })
+
+        response = self.client.get(url + '?download=1')
+        assert response.status_code == 200, response.content
```

Comment:
It'd be worth adding an assertion for the contents of the payload here to match the `BytesIO` contents to confirm that we've actually downloaded the correct file.

As well as the correct headers. `Content-Length`, `Content-Type`, and `Content-Disposition`. Mostly because in this case, these things are actually significant and we wouldn't want to break them.

Other than this, lgtm.